airflow/models.py

```diff
@@ -2875,22 +2956,40 @@ def evaluate(self, execution_date, session):
         :param execution_date: the date on which the trigger should be evaluated
         :type execution_date: datetime
         :return: whether the dependency is triggered or not and the reason
-        :rtype: (boolean, string)
+        :rtype: (boolean, Reason)
         """
-        states = self._states(execution_date, session)
+        states_by_schedule = self._states_by_schedule(execution_date,
+                                                      session)
```

Comment:
we could extract the rest of this method into another function that will receive `states_by_schedule` => that resulting function will then be easy to unit test